stream-chat-android-client/src/main/java/io/getstream/chat/android/client/notifications/LoadNotificationDataWorker.kt

```diff
@@ -0,0 +1,149 @@
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingWorkPolicy
+import androidx.work.ForegroundInfo
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import androidx.work.workDataOf
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.R
+import io.getstream.chat.android.client.logger.ChatLogger
+
+internal class LoadNotificationDataWorker(
+    private val context: Context,
+    workerParams: WorkerParameters,
+) : CoroutineWorker(context, workerParams) {
+
+    private val logger = ChatLogger.get("LoadNotificationDataWorker")
+
+    override suspend fun doWork(): Result {
+        val channelId: String = inputData.getString(DATA_CHANNEL_ID)!!
+        val channelType: String = inputData.getString(DATA_CHANNEL_TYPE)!!
+        val messageId: String = inputData.getString(DATA_MESSAGE_ID)!!
+        val notificationTitle: String = inputData.getString(DATA_NOTIFICATION_TITLE)!!
+        val notificationIcon: Int = inputData.getInt(DATA_NOTIFICATION_ICON, R.drawable.stream_ic_notification)
+        val notificationChannelName: String = inputData.getString(DATA_NOTIFICATION_CHANNEL_NAME)!!
+
+        setForeground(
+            createForegroundInfo(
+                notificationChannelId = LOAD_NOTIFICATION_DATA_CHANNEL_ID,
+                notificationChannelName = notificationChannelName,
+                notificationIcon = notificationIcon,
+                notificationTitle = notificationTitle,
+            )
+        )
+
+        return try {
+            ChatClient.displayNotificationWithData(channelId, channelType, messageId)
+            Result.success()
+        } catch (exception: IllegalStateException) {
+            logger.logE("Error while loading notification data: ${exception.message}")
+            Result.failure()
+        }
+    }
+
+    private fun createForegroundInfo(
+        notificationChannelId: String,
+        notificationIcon: Int,
+        notificationTitle: String,
+        notificationChannelName: String,
+    ): ForegroundInfo {
+        return ForegroundInfo(
+            NOTIFICATION_ID,
+            createForegroundNotification(
+                notificationChannelId = notificationChannelId,
+                notificationChannelName = notificationChannelName,
+                notificationIcon = notificationIcon,
+                notificationTitle = notificationTitle,
+            ),
+        )
+    }
+
+    private fun createForegroundNotification(
+        notificationChannelId: String,
+        notificationIcon: Int,
+        notificationTitle: String,
+        notificationChannelName: String,
+    ): Notification {
+        createSyncNotificationChannel(
+            notificationChannelId = notificationChannelId,
+            notificationChannelName = notificationChannelName,
+        )
+        return NotificationCompat.Builder(context, notificationChannelId)
```

Comment:
Should we allow to customize this notification? :thinking:

Reply:
In the current implementation, there is a possibility to customize icon and title. We can make it customizable but we should find a way to pass the whole `Notification` object to worker